fearless/Modules/BalanceInfo/BalanceInfoPresenter.swift

```diff
@@ -47,9 +53,46 @@ final class BalanceInfoPresenter {
 extension BalanceInfoPresenter: BalanceInfoViewOutput {
     func didLoad(view: BalanceInfoViewInput) {
         self.view = view
-        interactor.setup(with: self)
+        interactor.setup(with: self, for: balanceInfoType)
+    }
 
-        interactor.fetchBalance(for: balanceInfoType)
+    func didTapInfoButton() {
+        guard case let .chainAsset(wallet, chainAsset) = balanceInfoType,
+              let balance = balances[wallet.metaId] else {
+            return
+        }
+        if let accountId = wallet.fetch(for: chainAsset.chain.accountRequest())?.accountId,
+           let accountInfo = balance.accountInfos[chainAsset.uniqueKey(accountId: accountId)],
+           let info = accountInfo,
+           let free = Decimal.fromSubstratePerbill(value: info.data.free),
+           let reserved = Decimal.fromSubstratePerbill(value: info.data.reserved),
+           let miscFrozen = Decimal.fromSubstratePerbill(value: info.data.miscFrozen),
+           let feeFrozen = Decimal.fromSubstratePerbill(value: info.data.feeFrozen),
+           let minBalance = minimumBalance,
+           let decimalMinBalance = Decimal.fromSubstratePerbill(value: minBalance),
+           let locks = balanceLocks {
+            var price: Decimal = 0
+            let priceData = balance.prices.first(where: { $0.priceId == chainAsset.asset.priceId })
+            if let data = priceData, let decimalPrice = Decimal(string: data.price) {
+                price = decimalPrice
+            }
+            let balanceContext = BalanceContext(
```

Comment:
I can suggest to move making of BalanceContext to function

Reply:
and maybe it's make sense to set visibility of this button depending on this 'if let' construction result?